keras_nlp/models/bert.py

```diff
@@ -405,10 +419,66 @@ def BertBase(weights=None, vocabulary_size=None, name=None, trainable=True):
     return model
 
 
+def BertLarge(weights=None, vocabulary_size=None, name=None, trainable=True):
+
+    if (vocabulary_size is None and weights is None) or (
```

Comment:
Given that we are going to be doing this for BertMedium, BertSmall, etc. on a subsequent PR, it probably does make sense to refactor this a little.

My vote, rather than a big function that takes in every bert hyperparameter, would be to split off a function to validate and fetch the proper `weights` and `vocabulary_size`. Then the body of each of these can look something like.

```python
# Note that weights will be the local filepath after this call.
# All error messages can live in this function.
weights, vocabulary_size = _get_weights_and_vocab_size("bert_base", weights, vocabulary_size)

model = BertCustom(
    vocabulary_size=vocabulary_size,
    num_layers=24,
    num_heads=16,
    hidden_dim=1024,
    intermediate_dim=4096,
    dropout=0.1,
    max_sequence_length=512,
    name=name,
    trainable=trainable,
)

if weights:
    model.load_weights(weights)

return model
```